src/LinkedList.ts

```diff
@@ -0,0 +1,606 @@
+}
+
+/**
+ * Updates an element from a list at the specified index, returning a new
+ * list or `None` if the index is out-of-bounds.
+ * @since 2.1.1
+ */
+export function updateAt<A>(index: number, a: A, fa: LinkedList<A>): O.Option<LinkedList<A>> {
+  if (isNil(fa)) return O.none
+
+  let l: LinkedList<A> = fa
+  let i = 0
+  let out: LinkedList<A> = nil
+  let hasBeenUpdated = false
+  while (isCons(l)) {
+    if (i === index) {
+      hasBeenUpdated = true
+      out = cons(a, out)
+    } else {
+      out = cons(l.head, out)
+    }
+    l = l.tail
+    i++
+  }
+  return hasBeenUpdated ? O.some(reverse(out)) : O.none
+}
+
+/**
+ * Update the element at the specified index by applying a function
+ * to the current value, returning a new list or `None` if the index is out-of-bounds.
+ * @since 2.1.1
+ */
+export function modifyAt<A>(index: number, f: (a: A) => A, fa: LinkedList<A>): O.Option<LinkedList<A>> {
+  if (isNil(fa)) return O.none
+
+  let l: LinkedList<A> = fa
+  let i = 0
+  let out: LinkedList<A> = nil
+  let hasBeenUpdated = false
+  while (isCons(l)) {
+    if (i === index) {
+      hasBeenUpdated = true
+      out = cons(f(l.head), out)
+    } else {
+      out = cons(l.head, out)
+    }
+    l = l.tail
+    i++
+  }
+  return hasBeenUpdated ? O.some(reverse(out)) : O.none
+}
+
+/**
+ * Updates or deletes the element at the specified index by applying a function
+ * to the current value, returning a new list or `None` if the index is out-of-bounds.
+ * @since 2.1.1
+ */
+export function alterAt<A>(index: number, f: (a: A) => O.Option<A>, fa: LinkedList<A>): O.Option<LinkedList<A>> {
+  if (isNil(fa)) return O.none
+
+  let l: LinkedList<A> = fa
+  let i = 0
+  let out: LinkedList<A> = nil
+  let hasBeenAltered = false
+  while (isCons(l)) {
+    if (i === index) {
+      const oA = f(l.head)
+      if (O.isSome(oA)) {
+        out = cons(oA.value, out)
+      }
+      hasBeenAltered = true
+    } else {
+      out = cons(l.head, out)
+    }
+    l = l.tail
+    i++
+  }
+  return hasBeenAltered ? O.some(reverse(out)) : O.none
+}
+
+/**
+ * Reverse a list.
+ * @since 2.1.1
+ */
+export function reverse<A>(fa: LinkedList<A>): LinkedList<A> {
+  let out: LinkedList<A> = nil
+  let l = fa
+  while (isCons(l)) {
+    out = cons(l.head, out)
+    l = l.tail
+  }
+  return out
+}
+
+/**
+ * Flattens a list of lists.
+ * @since 2.1.1
+ */
+export function flatten<A>(mma: LinkedList<LinkedList<A>>): LinkedList<A> {
+  let out: LinkedList<A> = nil
+  let outer = mma
+  while (isCons(outer)) {
+    let inner = outer.head
+    while (isCons(inner)) {
+      out = cons(inner.head, out)
+      inner = inner.tail
+    }
+    outer = outer.tail
+  }
+  return reverse(out)
+}
+
+/**
+ * Sort the elements of a list in increasing order, where elements
+ * are compared using the specified ordering.
+ * @since 2.1.1
+ */
+export function sort<A>(O: Ord<A>): (fa: LinkedList<A>) => LinkedList<A> {
+  return flow(toArray, A.sort(O), fromArray)
```

Comment:
This thing walks the array twice (other than sort it). Perhaps makes sense to implement a custom algorithm?